src/client.ts

```diff
@@ -77,47 +77,48 @@ export default class UnleashClient extends EventEmitter {
     feature: FeatureInterface | undefined,
     context: Context,
     fallback: Function,
-  ): [boolean, VariantDefinition | undefined] {
+  ): StrategyResult {
     if (!feature) {
-      return [fallback(), undefined];
+      return { enabled: fallback() };
     }
 
     if (!feature || !feature.enabled) {
-      return [false, undefined];
+      return { enabled: false };
     }
 
     if (!Array.isArray(feature.strategies)) {
       const msg = `Malformed feature, strategies not an array, is a ${typeof feature.strategies}`;
       this.emit(UnleashEvents.Error, new Error(msg));
-      return [false, undefined];
+      return { enabled: false };
     }
 
     if (feature.strategies.length === 0) {
-      return [feature.enabled, undefined];
+      return { enabled: feature.enabled };
     }
 
-    let featureVariant = undefined;
+    let strategyResult = { enabled: false };
 
-    return [(
-      feature.strategies.length > 0 &&
+    feature.strategies?.some((strategySelector): boolean => {
```

Comment:
I left some in the impl since we want the early exit/short-circuit semantics

Reply:
Probably not super relevant, but is `some` guaranteed to traverse the list in order? Or is it just ... most likely to do that?